### DISCUSSION TITLE
pfelk with grafana

### DISCUSSION THREAD

**Discussion by nick2000713** [pfelk/pfelk#433] [General]
Hello, I am not sure if I am in the right place with this problem. I have used pfelk to display the data in grafana, at that time I had simply edited the elasticsearch.yml. But since the new security features of elastic search this does not work anymore. The grafana instans are not on the same server so I have to make elasticsearch accessible from outside. Can anyone help me or know anything about this? 

What I have changed in elasticsearch.yml :
- network.host: 0.0.0.0
- http.port: 9200
- discovery.seed_hosts: ["0.0.0.0"]

On the grafana site I tried
![Screenshot 2022-06-06 213858](https://user-images.githubusercontent.com/81333562/172234991-c6fff2e3-142d-4c51-b99a-d2dcbf8466c8.png)
 Basic auth and CA Cert. but I always get the error: Elasticsearch error: Bad Gateway

**Comment by a3ilson** [comment]
@b4b857f6ee had previously integrated grafana with pfelk. 

I’m unaware if grafana supports x pack security but if desired you can disable x pack security education as of v8.0.0 is turned on by default.

**Comment by nick2000713** [comment]
Thank you very much for your help. 
I was able to establish the connection and it works so far. 
I have deactivated xpack security for this
- xpack.security.enabled: false

But now Kibana no longer works and no new data comes from opnsense. do you have an idea? unfortunately I can't find any settings in kibana.yml and logstash.yml for xpack security.

**Comment by nick2000713** [comment]
NEWER ME:

I did it, it works now. 

CHANGES:

50-outputs.pfelk:

output 
-hosts => ["http://localhost:9200"]
-#hosts => ["https://localhost:9200"]
-# ssl => true
-#cacert => '/etc/logstash/config/certs/http_ca.crt'
 

elasticsearch.yml:

-network.host: 0.0.0.0
-http.port: 9200
-discovery.seed_hosts: ["0.0.0.0"]
-xpack.security.enabled: false
-xpack.security.enrollment.enabled: false
-xpack.security.http.ssl:
    enabled: false
    keystore.path: certs/http.p12
-xpack.security.transport.ssl:
    enabled: false
-http.host: 0.0.0.0
-transport.host: 0.0.0.0

kibana.yml:

-#elasticsearch.hosts: ['https://YOURHOST:9200']
-elasticsearch.hosts: ['http://YOURHOST:9200']
-elasticsearch.serviceAccountToken: BLABLABLA
-#elasticsearch.ssl.certificateAuthorities: [/var/lib/kibana/ca_1654537453963.crt]
-#xpack.fleet.outputs:

Thank you so much for your help, it's a really cool project.

**Comment by a3ilson** [comment]
@nick2000713 - Excellent and glad it worked out.
